rmqsrc/encode.nim

```diff
@@ -0,0 +1,102 @@
+import endians, tables, strutils
+import spec, frame
+
+# Encode basic types
+proc encode(v: uint8): array[0..0, char] =
+  return [v.char]
+
+proc encode(v: uint16): array[0..1, char] =
+  var v = v
+  bigEndian16(addr result, addr v)
+
+proc encode(v: uint32): array[0..3, char] =
+  var v = v
+  bigEndian32(addr result, addr v)
+
+proc encode(v: string): seq[char] =
+  result = newSeq[char]()
+  for ch in v:
+    result.add(ch)
+
+proc encode(v: Table[string, string]): seq[char] =
+  # TODO fill this out, right now we will just assume the table is empty
+  result = newSeq[char]()
+  for ch in encode(0.uint32):
+    result.add(ch)
+
+# Define encodings method parameters
+type 
+  ConnectionStartOk = tuple
+    index: uint32
+    clientProps: Table[string, string]
+    mechanism: (string, uint8) # short string
```

Comment:
Since these are encoded length-then-value, seems reasonable to order them that way in the type.